terraform/modules/provision-aws/eks.tf

```diff
@@ -295,6 +295,9 @@ resource "null_resource" "cluster-functional" {
     null_resource.prerequisite_binaries_present,
     module.eks.cluster_id,
     module.eks.eks_managed_node_groups,
+    module.eks.aws_eks_addon,
+    module.eks,
+    module.vpc
```

Comment:
I mean... isn't terraform's whole premise that it works better with less explicit directions haha... This is proving the antithesis to that haha..

Reply:
If we knew the exact dependencies we could squinch this down tighter; this is just the simplest way to address the problem without doing that additional research. 

That said, this also has the effect of soft-enforcing the dependency between the `provision-k8s` and the `provision-aws` modules, even when they're being called as one! So it's closer to the "safe" module layout that Terraform recommends.